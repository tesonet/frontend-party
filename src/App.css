.login {
  background: url(./img/background.png) no-repeat center center fixed;
  -webkit-background-size: cover;
  -moz-background-size: cover;
  -o-background-size: cover;
  background-size: cover;
  background-color: rgba(0, 0, 0, 0);
}

body {
  background-color: #0b0f27;
  font-family: 'Roboto', sans-serif;
}


.App {
  text-align: center;
}

.App-logo {
  width: 246px;
  pointer-events: none;
}

.App-header {
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}