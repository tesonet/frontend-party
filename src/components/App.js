import React from 'react';
import { Router, Route } from 'react-router-dom';
import history from './history';
import LoginPage from './Login/LoginPage';
import ResultPage from './Result/ResultPage';
import '../sass/app.scss';

const App = () => {
  return(
    <div className="new-app">
      <Router history={history}>
        <div>
          <Route path="/" exact component={LoginPage} />
          <Route path="/result" exact component={ResultPage} />
        </div>
      </Router>
    </div>
  );
};

export default App;
