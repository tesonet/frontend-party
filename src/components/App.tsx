import { hot } from 'react-hot-loader/root';
import React from 'react';
import {
  BrowserRouter,
  Switch,
  Route,
  Redirect,
} from 'react-router-dom';
import { Normalize } from 'styled-normalize';
import { ThemeProvider } from 'styled-components';
import GlobalStyle from './styled/GlobalStyle';
import Login from './pages/Login';
import ProtectedRoute from './utils/ProtectedRoute';
import ServerList from './pages/ServerList';
import theme from '../theme';
import { Provider } from 'react-redux';
import store from '../store';
import CheckIfLoggedIn from './utils/CheckIfLoggedIn';

function App() {
  return (
    <Provider store={store}>
      <CheckIfLoggedIn>
        <ThemeProvider theme={theme}>
          <Normalize />
          <GlobalStyle />
          <BrowserRouter>
            <Switch>
              <ProtectedRoute exact path="/" component={ServerList} />
              <Route exact path="/login" component={Login} />
              <Redirect to="/" />
            </Switch>
          </BrowserRouter>
        </ThemeProvider>
      </CheckIfLoggedIn>
    </Provider>
  );
}

export default hot(App);
