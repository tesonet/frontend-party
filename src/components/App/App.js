import React from 'react';
import PropTypes from 'prop-types';
import Helmet from 'react-helmet';
import { ThemeProvider } from 'styled-components';

import defaultTheme from '../../styles/defaultTheme';
import { app as copy } from '../../assets/copy/global.json';

export const App = props => (
  <ThemeProvider theme={defaultTheme}>
    <div>
      <Helmet
        defaultTitle={copy.defaultTitle}
        titleTemplate={copy.defaultTitleTpl}
        meta={[{ name: 'description', content: copy.defaultDescription }]}
      />
      {props.children}
    </div>
  </ThemeProvider>
);

App.propTypes = {
  children: PropTypes.node.isRequired,
};

export default App;
