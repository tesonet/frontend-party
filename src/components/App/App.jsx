import React from 'react';
import { Route, BrowserRouter as Router, Switch } from 'react-router-dom'
import Login from '../../containers/Login/Login';
import List from '../../containers/List/List';
import NotFound from '../NotFound/NotFound';
import PrivateRoute from '../PrivateRoute/PrivateRoute';

const App = () => {
  return (
    <Router>
      <Switch>
        {["/", "/login"].map(path => (
          <Route
            key={path}
            exact
            path={path}
            component={Login}
          />
        ))}
        <PrivateRoute exact path="/list" component={List} />
        <Route path="*" component={NotFound} />
      </Switch>
    </Router>
  )
}

export default App;