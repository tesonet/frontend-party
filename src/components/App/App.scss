.App {
  background-image: url('../assets/images/Background.png');
  background-position: center right;
  background-size: cover;
  bottom: 0;
  left: 0;
  position: absolute;
  right: 0;
  top: 0;
  z-index: -1;
}