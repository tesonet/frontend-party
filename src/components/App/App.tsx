import React from "react";
import {
  BrowserRouter as Router,
  Switch,
  Route,
  Redirect,
  RouteProps
} from "react-router-dom";
import Login from "../Login/Login";
import Servers from "../Servers/Servers";
import { Provider, useSelector } from "react-redux";
import store, { State } from "../../store/store";

const PrivateRoute: React.FC<RouteProps> = ({
  component: Component,
  ...rest
}) => {
  const token = useSelector(({ user }: State) => (user ? user.token : null));

  return (
    <Route
      {...rest}
      render={props =>
        token ? <Component {...props} /> : <Redirect to="/login" />
      }
    />
  );
};

const App: React.FC = () => (
  <Router>
    <Provider store={store}>
      <Switch>
        <Route exact path="/" render={() => <Redirect to="/login" />} />
        <Route path="/login" component={Login} />
        <PrivateRoute path="/servers" component={Servers} />
      </Switch>
    </Provider>
  </Router>
);

export default App;
