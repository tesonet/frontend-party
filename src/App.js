import React from 'react'
import {Provider} from 'react-redux'
import {Router, Route, Switch} from 'react-router-dom'

import {configureStore} from './store'
import {Login, Dashboard, NotFound} from './components/pages'
import PrivateRoute from './components/common/PrivateRoute'
import CssBaseline from './ui/CssBaseline'
import './App.css'

const {store, history} = configureStore()

function App() {
  return (
    <Provider store={store}>
      <CssBaseline />
      <Router history={history}>
        <Switch>
          <PrivateRoute exact path="/" component={Dashboard} />
          <Route exact path="/login" component={Login} />
          <Route path="*" component={NotFound} />
        </Switch>
      </Router>
    </Provider>
  )
}

export default App
