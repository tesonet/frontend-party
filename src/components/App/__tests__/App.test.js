import React from 'react';
import { shallow } from 'enzyme';

import App from '../App';

describe('App component', () => {
  const props = {
    children: <div>Child component</div>,
  };

  it('should render the app container with child components.', () => {
    const renderedComponent = shallow(<App {...props} />).dive();

    expect(renderedComponent.type()).toEqual('div');
    expect(renderedComponent).toMatchSnapshot();
  });
});
