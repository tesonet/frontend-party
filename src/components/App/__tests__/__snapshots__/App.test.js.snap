// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should render the app container with child components. 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ThemeProvider
    theme={
        Object {
            "bgColorDark": "#0b0f27",
            "bgColorPrimary": "#fff",
            "bgColorSecondary": "#f5f5f5",
            "bgColorTertiary": "#fdfdfd",
            "btnBgColorDisabled": "#b3b3b3",
            "btnBgColorHover": "#99cc33",
            "btnBgColorPrimary": "#9fd533",
            "btnTextColor": "#2f3254",
            "inputTextColorActive": "#b3b3b3",
            "inputTextColorPrimary": "#ccc",
            "listBorderColor": "#e6e6e6",
            "listTextColor": "#666666",
            "listTextColorHead": "#999",
          }
    }
>
    <div>
        <HelmetWrapper
            defaultTitle="Senior Frontend Party"
            defer={true}
            encodeSpecialCharacters={true}
            meta={
                Array [
                    Object {
                      "content": "Senior Frontend Party App - Testio.",
                      "name": "description",
                    },
                  ]
            }
            titleTemplate="SFP Testio | %s"
        />
        <div>
            Child component
        </div>
    </div>
</ThemeProvider>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <HelmetWrapper
          defaultTitle="Senior Frontend Party"
          defer={true}
          encodeSpecialCharacters={true}
          meta={
                    Array [
                              Object {
                                "content": "Senior Frontend Party App - Testio.",
                                "name": "description",
                              },
                            ]
          }
          titleTemplate="SFP Testio | %s"
/>,
        <div>
          Child component
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "defaultTitle": "Senior Frontend Party",
          "defer": true,
          "encodeSpecialCharacters": true,
          "meta": Array [
            Object {
              "content": "Senior Frontend Party App - Testio.",
              "name": "description",
            },
          ],
          "titleTemplate": "SFP Testio | %s",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Child component",
        },
        "ref": null,
        "rendered": "Child component",
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <HelmetWrapper
            defaultTitle="Senior Frontend Party"
            defer={true}
            encodeSpecialCharacters={true}
            meta={
                        Array [
                                    Object {
                                      "content": "Senior Frontend Party App - Testio.",
                                      "name": "description",
                                    },
                                  ]
            }
            titleTemplate="SFP Testio | %s"
/>,
          <div>
            Child component
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "defaultTitle": "Senior Frontend Party",
            "defer": true,
            "encodeSpecialCharacters": true,
            "meta": Array [
              Object {
                "content": "Senior Frontend Party App - Testio.",
                "name": "description",
              },
            ],
            "titleTemplate": "SFP Testio | %s",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Child component",
          },
          "ref": null,
          "rendered": "Child component",
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
